### PR TITLE
Support skipping control in Verifier

### PR DESCRIPTION
**Pull request by caithagoras**
```
== RELEASE NOTES ==

Verifier Changes
* Add support to skip running control queries and comparing results. This can be enabled by configuration property ``skip-control``.
```

### CODE REVIEW

File: presto-verifier/src/main/java/com/facebook/presto/verifier/event/VerifierQueryEvent.java [hunk 1]
```diff
@@ -113,10 +114,12 @@ public static VerifierQueryEvent skipped(
                 Optional.empty(),
                 Optional.empty(),
                 Optional.empty(),
-                new QueryInfo(
+                skipControl
+                        ? Optional.empty()
+                        : Optional.of(new QueryInfo(
                         sourceQuery.getControlConfiguration().getCatalog(),
```
Comment (wenleix) [inline presto-verifier/src/main/java/com/facebook/presto/verifier/event/VerifierQueryEvent.java:120]: nit: is the indentation suggested by Intellij?

File: presto-verifier/src/main/java/com/facebook/presto/verifier/framework/AbstractVerification.java [hunk 2]
```diff
@@ -176,22 +185,25 @@ public VerificationResult run()
...
+                    determinismAnalysis = Optional.of(determinismAnalyzer.analyze(control.get(), matchResult.get().getControlChecksum(), determinismAnalysisDetails));
+                }
             }
 
-            partialResult = Optional.of(concludeVerificationPartial(control, test, controlQueryContext, matchResult, determinismAnalysis, Optional.empty()));
+            partialResult = Optional.of(concludeVerificationPartial(control, test, controlQueryContext, testQueryContext, matchResult, determinismAnalysis, Optional.empty()));
         }
-        catch (Throwable t) {
+        catch (
+                Throwable t) {
```
Comment (wenleix) [inline presto-verifier/src/main/java/com/facebook/presto/verifier/framework/AbstractVerification.java:200]: nit: unnecessary line break?

File: presto-verifier/src/main/java/com/facebook/presto/verifier/framework/AbstractVerification.java [hunk 3]
```diff
@@ -204,18 +216,21 @@ public VerificationResult run()
...
-                determinismAnalysis,
-                controlChecksumQueryContext,
-                testChecksumQueryContext,
-                determinismAnalysisDetails.build(),
-                throwable);
+        return
+
+                concludeVerification(
+                        partialResult.get(),
+
```
Comment (wenleix) [inline presto-verifier/src/main/java/com/facebook/presto/verifier/framework/AbstractVerification.java:223]: nit: unnecessary empty line?

File: presto-verifier/src/main/java/com/facebook/presto/verifier/framework/AbstractVerification.java [hunk 4]
```diff
@@ -301,12 +319,14 @@ private VerificationResult concludeVerification(
                         Optional.of(determinismAnalysisDetails) :
                         Optional.empty(),
                 partialResult.getResolveMessage(),
-                buildQueryInfo(
+                skipControl
+                        ? Optional.empty()
+                        : Optional.of(buildQueryInfo(
                         sourceQuery.getControlConfiguration(),
```
Comment (wenleix) [inline presto-verifier/src/main/java/com/facebook/presto/verifier/framework/AbstractVerification.java:325]: nit: the indentation is not easy to read now...One idea is 
```java
: Optional.of(
      buildQueryInfo(
            sourceQuery.getControlConfiguration(),
............
```

But that's just my personal way. Feel free to keep the current approach if you see fit.

File: presto-verifier/src/main/java/com/facebook/presto/verifier/event/VerifierQueryEvent.java (hunk 1 above)
Comment (caithagoras) [reply presto-verifier/src/main/java/com/facebook/presto/verifier/event/VerifierQueryEvent.java:120]: It is

File: presto-verifier/src/main/java/com/facebook/presto/verifier/framework/AbstractVerification.java (hunk 4 above)
Comment (caithagoras) [reply presto-verifier/src/main/java/com/facebook/presto/verifier/framework/AbstractVerification.java:325]: Unfortunately IntelliJ auto-formats to the following if buildQueryInfo is on the next line:
```
skipControl
                        ? Optional.empty()
                        : Optional.of(
                        buildQueryInfo(
                                sourceQuery.getControlConfiguration(),
                                sourceQuery.getControlQuery(),
                                controlChecksumQueryContext,
                                control,
                                controlQueryContext)),
```

Comment (caithagoras) [reply presto-verifier/src/main/java/com/facebook/presto/verifier/framework/AbstractVerification.java:325]: Moving the ternary operator to the end of the line and indentation works better.

File: presto-verifier/src/main/java/com/facebook/presto/verifier/event/VerifierQueryEvent.java (hunk 1 above)
Comment (caithagoras) [reply presto-verifier/src/main/java/com/facebook/presto/verifier/event/VerifierQueryEvent.java]: Resolved by moving ternary operator to the end of the line.

**Pull request merged**